05-JWT-Basics/starter/controllers/main.js

```diff
@@ -0,0 +1,37 @@
+
+const jwt = require('jsonwebtoken')
+
+const {BadRequestError} = require('../errors')
+
+
+
+const login = async (req,res) => {
+    const {username, password} = req.body;
+
+    //check in the controller
+    if(!username || !password) {
+        throw new BadRequestError('Please provide email and password')
```

Comment:
Minor note: there is a mismatch here. It asks for "username", but the error text says "email". Not really an error, but something to keep in mind for the future

Reply:
Thank you so much for your help.